plonk-core/src/lookup/lookup_table.rs

```diff
@@ -0,0 +1,368 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::error::Error;
+use crate::lookup::MultiSet;
+use ark_ff::Field;
+
+/// This struct is a table, contaning a vector, of arity 4 where each of the
+/// values is a scalar. The elements of the table are determined by the function
+/// g for g(x,y), used to compute tuples.
+///
+/// This struct will be used to determine the outputs of gates within arithmetic
+/// circuits.
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct LookupTable<F>(pub Vec<[F; 4]>)
+where
+    F: Field;
+
+impl<F> LookupTable<F>
+where
+    F: Field,
+{
+    /// Create a new, empty Plookup table, with arity 4.
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Pushes a row to the `LookupTable` vector.
+    fn push(&mut self, row: [F; 4]) {
+        self.0.push(row);
+    }
+
+    /// Returns the length of the `LookupTable` vector.
+    pub fn size(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Insert a new row for an addition operation.
+    /// This function needs to know the upper bound of the amount of addition
+    /// operations that will be done in the plookup table.
+    pub fn insert_add_row(&mut self, a: u64, b: u64, upper_bound: u64) {
+        let c = (a + b) % upper_bound;
+        self.push([F::from(a), F::from(b), F::from(c), F::zero()]);
+    }
+
+    /// Insert a new row for an addition operation.
+    /// This function needs to know the upper bound of the amount of addition
+    /// operations that will be done in the plookup table.
+    pub fn insert_special_row(&mut self, a: F, b: F, c: F, d: F) {
+        self.push([a, b, c, d]);
+    }
```

Comment:
The comment is misleading. I would also just call the name of the method `insert_row`.

Reply:
Renamed to `insert_row` and rewrote other `insert_{}_row` functions to call it